src/features/mangaViewer/components/LanguageModel.tsx

```diff
@@ -0,0 +1,127 @@
+import { useQuery } from "@tanstack/react-query";
+import { getComicChapters } from "../../../sharedAPI.ts/apiQueries";
+import { MangaDetails } from "../../../models/mangaDetails";
+import * as yup from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Resolver, useForm } from "react-hook-form";
+import { useReadStore } from "../../../state/readStore";
+
+const LanguageModal = ({ mangaInfo }: { mangaInfo: MangaDetails | null }) => {
+  const setCurrentComicLanguage = useReadStore(
+    (state) => state.setCurrentComicLanguage
+  );
+
+  const { isPending, data } = useQuery({
+    queryKey: ["comicChapter", mangaInfo?.comic?.hid],
+    queryFn: () => getComicChapters(mangaInfo?.comic?.hid),
+  });
+
+  const lang = new Intl.DisplayNames(["en"], { type: "language" });
+
+  const languages = [
+    ...new Map(
+      data?.chapters.map((chapter) => [lang.of(chapter.lang), chapter])
+    ).values(),
+  ];
+  let schemaObject = Object.fromEntries(
+    languages.map((field) => [field.lang, yup.string()])
+  );
+
+  const schema = yup.object().shape(schemaObject);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<{ [key: string]: string }>({
+    resolver: yupResolver(schema) as Resolver<{ [key: string]: string }>,
+  });
+
+  function setLangData(data: { [key: string]: string }) {
+    setCurrentComicLanguage(data.lang);
+
+    const modal = document.getElementById("lang_modal") as HTMLDialogElement;
```

Comment:
This is very old school and not really react like. I would suggest doing some kind of short circuit instead.